File: project/jimmer-core/src/main/java/org/babyfish/jimmer/jackson/ImmutableProps.java
package org.babyfish.jimmer.jackson;

import org.babyfish.jimmer.meta.ImmutableProp;
import org.babyfish.jimmer.meta.ImmutableType;

import java.lang.reflect.Method;

/**
 * @deprecated 请使用 {@link org.babyfish.jimmer.json.ImmutableProps}。
 */
@Deprecated
public class ImmutableProps {

    private ImmutableProps() {
    }

    public static ImmutableProp get(ImmutableType type, Method method) {
        return org.babyfish.jimmer.json.ImmutableProps.get(type, method);
    }
}

File: project/jimmer-sql-kotlin/src/main/kotlin/org/babyfish/jimmer/sql/kt/util/ImmutableProps.kt
package org.babyfish.jimmer.sql.kt.util

import org.babyfish.jimmer.meta.ImmutableProp
import org.babyfish.jimmer.meta.ImmutableType
import kotlin.reflect.KProperty1
import kotlin.reflect.jvm.javaMethod

fun immutableProp(prop: KProperty1<*, *>): ImmutableProp {
    val javaMethod = prop.getter.javaMethod ?: error("$prop does not has java getter")
    return ImmutableType.get(javaMethod.declaringClass).getProp(prop.name)
}